### PR TITLE
Add a gflag to not canonicalize UUIDs

### PR DESCRIPTION
**Pull request by anirbanr-fb**
Summary: thie enables compatibility with UUIDs from mysql

Test Plan: Enhanced the unit test to toggle the gflag

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: src/kudu/util/oid_generator.cc [hunk 1]
```diff
@@ -31,15 +32,23 @@
 using std::string;
 using strings::Substitute;
 
+DEFINE_bool(cononicalize_uuid, true,
+            "If set this strips -/dashes from all uuids and makes "
```
Comment (mpercy) [inline src/kudu/util/oid_generator.cc:36]: nit: mind adding "at generation time" to this description?

File: src/kudu/util/oid_generator.cc [hunk 2]
```diff
@@ -31,15 +32,23 @@
 using std::string;
 using strings::Substitute;
 
+DEFINE_bool(cononicalize_uuid, true,
```
Comment (mpercy) [inline src/kudu/util/oid_generator.cc:35]: typo: canonicalize, also let's call it canonicalize_uuids

I know this is a little pedantic but we could probably upstream this patch

File: src/kudu/util/oid_generator-test.cc [hunk 3]
```diff
@@ -26,9 +26,12 @@
 
 using std::string;
 
+DECLARE_bool(cononicalize_uuid);
+
 namespace kudu {
 
 TEST(ObjectIdGeneratorTest, TestCanoicalizeUuid) {
+  FLAGS_cononicalize_uuid = true;
```
Comment (mpercy) [inline src/kudu/util/oid_generator-test.cc:35]: Before you change the flag, add:

    google::FlagSaver saver;

See hybrid_clock-test.cc for an example

File: src/kudu/util/oid_generator-test.cc [hunk 4]
```diff
@@ -49,4 +52,20 @@ TEST(ObjectIdGeneratorTest, TestCanoicalizeUuid) {
   ASSERT_EQ(kExpectedCanonicalized, canonicalized);
 }
 
+TEST(ObjectIdGeneratorTest, TestNoCanoicalizeUuid) {
+  FLAGS_cononicalize_uuid = false;
```
Comment (mpercy) [inline src/kudu/util/oid_generator-test.cc:58]: FlagSaver (see above)

**Review by mpercy — APPROVED** [review]
Looks good, I just have a few minor suggestions.

File: src/kudu/util/oid_generator.cc (hunk 2 above)
Comment (anirbanr-fb) [reply src/kudu/util/oid_generator.cc]: I am looking for someone better at this to do it.

**Pull request merged**